src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1652,6 +1651,8 @@ BlockIO InterpreterCreateQuery::execute()
 {
     FunctionNameNormalizer().visit(query_ptr.get());
     auto & create = query_ptr->as<ASTCreateQuery &>();
+    getContext()->setAttachQuery(create.attach);
+    SCOPE_EXIT(getContext()->setAttachQuery(false));
```

Comment:
Because we pass mutable context to `InterpreterCreateQuery` constructor and we can reuse this context later after we executed it, so we should set this flag only for `InterpreterCreateQuery::execute()`